File: parse_mhtml.mjs
import fs from 'fs';
import { parse } from 'node-html-parser';

const files = fs.readdirSync('scormstack');
for (const f of files) {
  if (!f.endsWith('.mhtml')) continue;
  const content = fs.readFileSync('scormstack/' + f, 'utf8');
  
  // Extract HTML base64 or quoted-printable
  let html = '';
  const lines = content.split('\n');
  let inHtml = false;
  let buffer = [];
  
  for (const line of lines) {
    if (line.includes('Content-Type: text/html')) {
      inHtml = true;
      buffer = [];
      continue;
    }
    if (inHtml && line.startsWith('------MultipartBoundary')) {
      inHtml = false;
      break;
    }
    if (inHtml && !line.includes('Content-Transfer-Encoding') && !line.includes('Content-Location')) {
      buffer.push(line.replace(/=$/, '').trim());
    }
  }
  
  const rawHtml = buffer.join('').replace(/=([0-9A-F]{2})/g, (match, p1) => String.fromCharCode(parseInt(p1, 16)));
  
  const outPath = 'scormstack/' + f + '.txt';
  fs.writeFileSync(outPath, rawHtml);
  console.log('Saved raw HTML for', f);
}

File: parse_mhtml.py
import email
import os
from bs4 import BeautifulSoup

d = 'scormstack'
for f in os.listdir(d):
    if not f.endswith('.mhtml'): continue
    path = os.path.join(d, f)
    with open(path, 'r', encoding='utf-8', errors='ignore') as fp:
        msg = email.message_from_file(fp)
        for part in msg.walk():
            if part.get_content_type() == 'text/html':
                html = part.get_payload(decode=True).decode('utf-8', errors='ignore')
                soup = BeautifulSoup(html, 'html.parser')
                
                # Remove scripts, styles
                for script in soup(["script", "style", "svg", "path", "img", "meta", "link"]):
                    script.extract()
                
                text = "\n".join([line.strip() for line in soup.get_text().splitlines() if line.strip()])
                out_path = path + '.txt'
                with open(out_path, 'w', encoding='utf-8') as out:
                    out.write(text)
                print(f"Parsed {f} -> {out_path}")
                break
